From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ted Meyer <tmathmeyer@chromium.org>
Date: Sun, 11 Oct 2026 08:42:51 -0700
Subject: [M152] [MultiBuffer] always post StopLoader

Original change's description:
> [MultiBuffer] always post StopLoader
>
> Stop can be called re-entrantly, from places which make use of weak
> references to `reader_`, so we need to post the destruction back onto
> this same task queue.
>
> Fixed: 565742182
> Change-Id: Ibaac92b7e77718379e5679fba3d37428d93f03b3
> Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/8465025
> Commit-Queue: Dale Curtis <dalecurtis@chromium.org>
> Auto-Submit: Ted (Chromium) Meyer <tmathmeyer@chromium.org>
> Reviewed-by: Dale Curtis <dalecurtis@chromium.org>
> Cr-Commit-Position: refs/heads/main@{#1706435}

(cherry picked from commit 8f54d92feff3932ab984565b6e71a356d0590797)

Bug: 567398389,565742182
Change-Id: Ibaac92b7e77718379e5679fba3d37428d93f03b3
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/8547408
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Auto-Submit: chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com <chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com>
Commit-Queue: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Cr-Commit-Position: refs/branch-heads/7977@{#2823}
Cr-Branched-From: b7fe14017379ddffae396d944fb8b59a5896c261-refs/heads/main@{#1669021}

diff --git a/third_party/blink/renderer/platform/media/multi_buffer_data_source.cc b/third_party/blink/renderer/platform/media/multi_buffer_data_source.cc
index 7925efb563e3b404884b3d3ab9aebd67c76a1ac9..1066b774840f571d14504c034fc56a18a6eaa631 100644
--- a/third_party/blink/renderer/platform/media/multi_buffer_data_source.cc
+++ b/third_party/blink/renderer/platform/media/multi_buffer_data_source.cc
@@ -355,13 +355,6 @@ void MultiBufferDataSource::Stop() {
   {
     base::AutoLock auto_lock(lock_);
     StopInternal_Locked();
-
-    // Cleanup resources immediately if we're already on the right thread.
-    if (render_task_runner_->BelongsToCurrentThread()) {
-      reader_.reset();
-      url_data_.reset();
-      return;
-    }
   }
 
   PostCrossThreadTask(*render_task_runner_, FROM_HERE,
